WordPress/gutenberg
# Add isRawAttribute to entity config

## Description

Solves https://github.com/WordPress/gutenberg/issues/34335

As seen in https://github.com/WordPress/gutenberg/issues/34335, core-data always assumes that entity records fields with a `raw` attribute are always "flattened" by `getRawEntityRecord`.

This behavior is useful for posts: `content.raw` in a post or a page refers to HTML with block markup as opposed to `content.rendered` which is the HTML without block markup.

Unfortunately, for other entity records such as widgets, this behavior is undesired.

To remedy the problem, this PR adds an explicit entity configuration field called `rawAttributes` which may be used to specify which fields should be treated as block markup HTML.

The downside of this approach is a BC break so I wonder if a safer solution here would be to do a "denylist" instead of an "allowlist" (e.g. `nonRawAttributes`).

## How has this been tested?
* Confirm the tests are green
* Create a new post, save it, change the contents, save again, confirm everything worked as expected
* Go to the widgets editor (`/wp-admin/widgets.php`), update a single widget, click "Update", confirm that the batch request only contained a single child request related to the updated widget (as opposed to one per widget).

## Types of changes
New feature (non-breaking change which adds functionality)
